Fix mypy findings

Solmath opened the pull request:
I was about to continue with the equalizer implementation, but then thought it would be a good idea to first get the mypy hook working.

First I fixed the types (admittedly with some AI support, so please have a closer look). Then I noticed some errors when running the examples. To fix these, I added logic to disconnect() to cancel all subscription tasks and close the session.

I bumped the version to 2.11.1, because there were no functional changes. Furthermore I refactored the pyproject.toml to use [project] instead of [tool.poetry] as suggested by PEP 621.

Finally I added the CXN 100 to the readme and updated the example code to match my trivial changes in the example files.

noahhusby commented:
Hi @Solmath,

Thanks for your contributions! Let's split this into a few separate PRs to make review easier: 
1) The basic mypy errors in the main library
2) PEP 621 Compliance
3) Documentation Update
4) Logic fixes for shutdown

In general, I don't covering multiple things in a single PR in case we need to roll something back. Looking forward to reviewing once split 👍

Solmath commented:
Thanks for the input, I totally agree. This PR somehow took on a life of its own once I started.

I'll split it up and create some more PRs.

Solmath commented:
Moved changes to new PRs as suggested and reverted changes here.

1. The basic mypy errors in the main library -> this PR
1. PEP 621 Compliance -> #149
1. Documentation Update -> #150
1. Logic fixes for shutdown -> #151

noahhusby commented on pyproject.toml:
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "aiostreammagic"
-version = "2.11.0"
```
Remove the bump. I'll handle the release after the PRs are merged.

noahhusby commented on aiostreammagic/stream_magic.py:
```diff
@@ -94,17 +94,20 @@ async def connect(self) -> Any:
-                await self.connect_task
-            except asyncio.CancelledError:
-                pass
+            if self.connect_task is not None:
```
```suggestion
            if self.connect_task:
```

noahhusby reviewed (commented):
Sorry for the delay in reviewing this. I will get this approved today.
